types/mongodb/index.d.ts

```diff
@@ -2140,14 +2142,14 @@ export interface FindOperatorsUnordered {
 }
 
 /** http://mongodb.github.io/node-mongodb-native/3.1/api/Collection.html#findOne */
-export interface FindOneOptions {
+export interface FindOneOptions<T> {
     limit?: number;
-    sort?: any[] | object;
-    projection?: object;
+    sort?: any[] | {[P in keyof T]?: number};
```

Comment:
Just like with projection, `sort` should accept input such as `{ 'field.nested': true }`.

Reply:
Thanks for the review!

I'm checked documentation and added more type definitions for projection and sort.   

The modified types of projection and sort only for `find` action, so it should have no influence to aggregation.

Reply:
Good job, though there are still some minor changes. While not explicitly documented, when passing sort input as an array of `[key, value]`, value could be any of `1, -1, 'asc', 'desc', 'ascending', 'descending'` in any case, i.e. `'AsC', 'dESC'` https://github.com/mongodb/node-mongodb-native/blob/81684319fe7ed4c89999983d7052ca1029d3f18c/src/utils.ts#L20

I've also added a comment about tests, to make sure these changes don't break more complex projections.